### ISSUE TITLE
Excluding Parameters/Bound Properties and Performance

### ISSUE THREAD

**Issue by TorreyGarland** [sleeuwen/UrlActionGenerator#8]
Is there anyway to have parameters excluded, either via individual attributes per parameter or an assembly level attribute.

I have several page model methods that either have a CancellationToken as a parameter or a custom type (GridParameters used for getting Grid Pagination and Sorting info using a custom ModelBinder).  I have run into some issues with the code that is generated for these parameters.

Also, and I think this is more due to the "new-ness" of source generators and interoperability with Razor Pages and not UrlActionGenerator, but editing both the C# code-behind and razor pages becomes dog-slow after installation.

**Comment by sleeuwen** [comment]
Hi @TorreyGarland,

Sounds like a good idea to add the ability to ignore specific types for the generated methods. In fact, I'm already ignoring some types like IFormFile. To exclude custom types, I think adding an assembly attribute would be the easiest way to do it. Would something like this for you? in that case any parameters of the type `GridParameters` will not be included in the generated method.
```C#
[assembly: UrlActionGenerator.ExcludeUrlParameterType(typeof(GridParameters))]
```
I'll probably add CancellationToken as a default excluded parameter type as well.

Regarding the performance, the source generator is invoked a lot while editing so even if it only takes a few more milliseconds it can degrade the experience a lot. The integration of source generators and razor pages is currently non-existent (as far as I know) so for that I have to resort to file system reads, together with the fact it is invoked a lot it can become very slow. Having a quick look at the new razor source generator for .NET 6 it looks like they are using a new incremental generator, that could help improve this in the future but I haven't seen any documentation about the changes in .NET 6 for this. I will keep an eye on this and update the generator to use the new features when possible.
For now, I'll try to see if I can reduce the time it takes for the source generator to complete.

**Comment by TorreyGarland** [comment]
Thanks for looking into this!

**Comment by JoaoFCorreia** [comment]
Hi @TorreyGarland the feature for ignoring parameters of type CancellationToken is still being planned?

**Comment by sleeuwen** [comment]
Hi @JoaoFCorreia I have actually implemented most of the changes to support ignoring custom types (and added CancellationToken as ignored by default). I will try to finish it either this week or next week and release a new version with the change.

**Comment by TorreyGarland** [comment]
have you considered using an IIncrementalGenerator and blazor capable?

**Comment by sleeuwen** [comment]
Hi @TorreyGarland yes I have looked at the IIncrementalGenerator and it is actually already released in version 0.5.0. I haven't looked into support for blazor yet

**Comment by sleeuwen** [comment]
@JoaoFCorreia I have just released a new version to nuget, 0.6.0, which now ignores CancellationToken parameters by default.

@TorreyGarland This release also enabled you to specify types that should not be generated as parameters by using an assembly attribute like this:
```C#
[assembly: UrlActionGenerator.ExcludedParameterType(typeof(MyCustomTypeThatShouldBeExcluded))]
```

**Comment by TorreyGarland** [comment]
Thanks again.
